sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
It can't, the Complex class would have to literally convert to an Add. And
of course, symengine Add can't hold 1+I, you would get an assert error in
debug mode. So what you are proposing is my proposal on the wiki, i.e.
convert everything to sympy before returning. And that would indeed work,
but would be slow until we port everything.

So my alternative proposal anove is better I think.

Sent from my mobile phone.
On Jul 1, 2016 4:33 PM, "Aaron Meurer" <notifications@github.com> wrote:

> In sympy/physics/mechanics/functions.py
> <https://github.com/sympy/sympy/pull/11002#discussion_r69362963>:
>
> > @@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
> >
> >  def _fraction_decomp(expr):
> >      """Return num, den such that expr = num/den"""
> > -    if not isinstance(expr, Mul):
> > +    if not isinstance(expr, MulClass):
>
> Couldn't Complex pretend to be an Add, for the purposes of its SymPy
> methods (like .args)? I suppose an Add with a complex would also need to
> pretend that it is a part of it (like Add(1, I, x).args should give (1,
> I, x), not (Complex(1+I), x)).
>
> That's the sort of thing that the wrapper would have to do, fake the SymPy
> API everywhere. If the SymPy API is bad, say, because it forces a slowdown,
> we can discuss if it should be changed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sympy/sympy/pull/11002/files/4c8b2d16e1820a104c81993c5905763e6698844c#r69362963>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABQWLIShxJ2IKoPA_iLjQmPhZDCP0d-ks5qRZWigaJpZM4IF-EF>
> .
>